improver/generate_ancillaries/generate_derived_solar_fields.py

```diff
@@ -291,7 +431,7 @@ def _create_solar_radiation_cube(
         attrs = generate_mandatory_attributes([target_grid])
 
         solar_radiation_cube = Cube(
```

Comment:
The acceptance test data suggest that this is not setting all the required meta-data. It isn't obvious to me what the meta-data should be in this instance. From the inputs, I get the impression that this Plugin is generating ancillary data, not forecast data, but I'm not sure.

1. Is the Linke Turbidity an ancillary file or one that is based on NWP forecasts? If the latter, it needs to have a time, forecast_period and forecast_reference_time. There's also no source attribute, but that might be ok.
2. The output KGO do not inherit institution and title attributes, and I suspect they should.
3. Should orography be a compulsory argument? I ask because I can't think of a reason why you'd want to run this algorithm without an orography field.
4. The orography file has no source, institute or title either - the data have come from somewhere and should be described. I suspect that this should be the origin of the same attributes in the output data if orography really should be compulsory.

Reply:
Hi @MoseleyS, sorry for the delay in reply. It was a long weekend over hear and had a few other priorities to address yesterday.

Thanks for raising this point, this is certainly a gap in this CLI. I haven't made any changes as yet, but will respond to your queries and try to rectify the issues tomorrow.

> From the inputs, I get the impression that this Plugin is generating ancillary data, not forecast data, but I'm not sure.

Yes, this Plugin is generating ancillary data, not forecast data.

Reply:
> 1. Is the Linke Turbidity an ancillary file or one that is based on NWP forecasts?

Here Linke turbidity provided is as an ancillary, so is not based on NWP forecasts.

In this instance the cube provided is one I've mocked up that has linearly varying linke turbidity which decreases with decreasing latitude (latitudes being negative in the SH, means smaller value at higher latitudes in the absolute sense). I've added a title to this cube to try and reflect that, but I could elaborate here to make that clearer.

An aside: More generally one can source or determine a climatological values for linke turbidity from observational data (satellite and ground based data), or evaluate estimates of linke turbidity value from other parameters (precipitable water and aerosol optical depth values), so one could generate linke turbidity based on NWP data. The following reference provides some details on this (under the section Linke Turbidity models).

For our usage (at least in the first instance), we've chosen to work with a constant value as it is relative values are sufficient. If one were to use outputs to be accurate measures of clearsky solar radiation, such as what is required for solar engineering purposes, one would need to source a more realistic dataset for linke turbidity.

> There's also no source attribute, but that might be ok.

This is something I can add but given this is mocked up data, what would you suggest? The alternative here would be to leave source out.

Reply:
> 2.	The output KGO do not inherit institution and title attributes, and I suspect they should.

I think this is due to these values being absent from the target grid. More generally though, I think some of these attributes should be generate in-place to better describe the output.

What makes sense for what these fields are the following:
1.	`institution` should be sourced from the target grid.
2.	`source` should be IMPROVER, given this fields is being generated by improver.
3.	`title` should be describe what the data is, so should use value defined in place rather than source this from input target-grid.

Does this sound reasonable?

Reply:
> 3. Should orography be a compulsory argument? I ask because I can't think of a reason why you'd want to run this algorithm without an orography field.

I've deliberately left orography as an optional argument.

Our intended use for this output is as a feature variable for the RainForests calibration. Here this variable (amongst others) are used to differentiate between weather situations to allow for situation dependent calibration to be applied on a per realization basis.

What this field provides us in the context is a measure of the maximum incoming solar energy into the associated grid-box. This parameter (unlike the others) has a direct correlation with latitude and time of year, so provides a measure of differentiation both spatially and seasonally.

While I agree that incorporating orography would provide a more realistic measure of the for this parameter around topography, it also brings in an additional correlation to the mix. While this may prove useful (allowing us to differentiate elevated areas), it may interact with the spatial/seasonality this parameter is implicitly providing. So in the first instance we have elected to keep things simple and to use this parameter without orography and consider clearsky solar radiation at a common altitude. In future we may well use orography as input, but this will require careful evaluation.

If one were after accurate measure of clearsky solar radiation, then much like linke turbidity, one should be using orography in the calculation.

Based on this I'm inclined to leave this as is.

Reply:
4.	The orography file has no source, institute or title either - the data have come from somewhere and should be described.

The orography data I used here was sourced from our ancillary dataset, and this file seems to lack these attributes. I've had a look at other similar ancillary inputs used in other acceptance tests (`apply-lapse-rate` for instance) and they also seem to lack these attributes. But to me it seems like these attributes should be present, even in these ancillaries.

This is something we should probably rectify, but there are higher priorities at the moment. So the simplest solution here would be to add these attributes to orography cube.

> I suspect that this should be the origin of the same attributes in the output data if orography really should be compulsory.

I think what makes most sense to me is that the attributes are sourced from the target grid. In the acceptance tests I've used orography as the target grid, so adding these attributes here should allow these to carry through to the outputs.

Does this sound reasonable to you?

Reply:
Institute, source and title have specific meanings described in https://metoffice.sharepoint.com/sites/IMPROVERMetOfficeCollaborationExt/SitePages/Metadata.aspx so I think title should be something like `"IMPROVER Ancillary on UK 2 km Standard Grid"` with the second half of this using the same wording you apply to your standard grid.

I'm happy that orography is an optional argument given your explanation of how you intend to use this.

I'm happy with your proposal. 

As a simple extra thing, please could you add the cube names of your new ancillaries to this list, so that all our acceptance test data continues to pass our metadata conventions.
https://github.com/benowen-bom/improver/blob/735d4faaa582362a35f8a2466b4e18f549a0da9b/improver/developer_tools/metadata_interpreter.py#L62

Reply:
Thanks @MoseleyS! I appreciate the link to the document, I found that really helpful.

I've updated `_create_solar_radiation_cube` to explicitly handle `source`, `title` attributes, with `institution` sourced from `target_grid` (or set as `"unknown"`) via `_create_solar_radiation_cube`. I've tried to get grid information from the `target_grid` cube, but to do this I've made an assumption on the format of title (namely that grid info follows on from the final instance of `"on "` present in title). If you're not happy with this approach, then I propose that we set `title` to a fixed string such as this:
```
attrs["title"] = "IMPROVER ancillary"
```
or:
```
attrs["title"] = "IMPROVER ancillary: accumulated clearsky solar radiation derived using Ineichen-Perez model."
```

Reply:
Also, @tjtg has sent through updated acceptance test data.